engine.go

```diff
@@ -168,11 +171,19 @@ func (e *engine) render() {
 func (e *engine) write() {
 	if *e.env.getArgs().Eval {
 		fmt.Printf("PS1=\"%s\"", e.renderer.string())
-		if e.rprompt != "" && e.env.getShellName() == zsh {
+		if e.env.getShellName() == zsh {
```

Comment:
I fixed a bug here with zsh. If the rprompt block had output once, and then later didn't have output, the zsh RPOMPT would never get cleared. An easy way to repro this is to create a block with a single text element, get a prompt, and then remove this block from your template and save it. Another way would be using the executiontime segment, which can sometimes emit output and sometimes not, depending on the threshold.

Reply:
Got it, didn't think of that!